tests/test_project.py

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env python3
+
+"""Test DUNE Version
+
+This script tests work with the a smart contract project keys:
+  --create-cmake-app
+  --create-bare-app
+  --cmake-build
+  --ctest
+  --gdb
+"""
+
+import glob
+import os
+import shutil
+import subprocess
+
+from common import DUNE_EXE,TEST_PATH
+
+
+PROJECT_NAME = "test_app"
+TEST_APP_DIR = TEST_PATH + "/" + PROJECT_NAME
+TEST_APP_BLD_DIR = TEST_APP_DIR + "/build/" + PROJECT_NAME
+TEST_APP_WASM = TEST_APP_BLD_DIR + "/" + PROJECT_NAME + ".wasm"    # TEST_APP_BLD_DIR + "/test_app.wasm"
+
+
+def remove_existing():
+    """ Remove an existing `./test_app` dir. """
+
+    if os.path.exists(TEST_APP_DIR):
+        print("Removing TEST_APP_DIR: ", TEST_APP_DIR)
+        shutil.rmtree(TEST_APP_DIR)
+
+
+def test_create_cmake_app():
+    """Test `--create-cmake-app` key."""
+
+    remove_existing()
+
+    # Expected files.
+    filelist = [TEST_APP_DIR + '/',
+                TEST_APP_DIR + '/src',
+                TEST_APP_DIR + '/src/' + PROJECT_NAME + '.cpp',
+                TEST_APP_DIR + '/src/CMakeLists.txt',
+                TEST_APP_DIR + '/include',
+                TEST_APP_DIR + '/include/' + PROJECT_NAME + '.hpp',
+                TEST_APP_DIR + '/ricardian',
+                TEST_APP_DIR + '/ricardian/' + PROJECT_NAME + '.contracts.md',
+                TEST_APP_DIR + '/build',
+                TEST_APP_DIR + '/CMakeLists.txt',
+                TEST_APP_DIR + '/README.txt']
+
+    # Create the test app.
+    subprocess.run([DUNE_EXE, "--create-cmake-app", PROJECT_NAME, TEST_PATH], check=True)
+    assert os.path.isdir(TEST_APP_DIR) is True
```

Comment:
In my case this fails (this directory doesn't exist yet?).
I think there might be race condition, i.e. line 55 executes before command from line 54 finishes?
Probably you should use `subprocess.check_output` after line 54 to wait for the output.

Reply:
I don't believe this is a blocking call. non-the-less, I've added code to look at the return value.

See 0e7f7d29da9f97fe4d9c0ef0f8e88fdc3ee5551e

Reply:
Interesting, it still fails on my side on checking if the directory exist.
When I run only `test_create_bare_app` it is fine. But if previously `test_create_cmake_app` is executed it fails.

Actually I can reproduce the issue when I run following command:
`/home/mikel/repo/DUNE/dune --create-cmake-app test_app /home/mikel/repo/DUNE/tests && rm -rf /home/mikel/repo/DUNE/test_app && /home/mikel/repo/DUNE/dune --create-bare-app test_app /home/mikel/repo/DUNE/tests`
It seems the run with `create-bare-app` doesn't create `test_app`.
Maybe we should create a separate issue for it.

Reply:
Create a new issue for this:
https://github.com/AntelopeIO/DUNE/issues/85